catch/include/performance_common.hh

```diff
@@ -0,0 +1,246 @@
+  float& time_;
+  hipStream_t stream_;
+};
+
+class EventTimer : public Timer {
+ public:
+  EventTimer(float& time, hipStream_t stream = nullptr) : Timer(time, stream) {
+    HIP_CHECK(hipEventCreate(&start_));
+    HIP_CHECK(hipEventCreate(&stop_));
+    HIP_CHECK(hipEventRecord(start_, GetStream()));
+  }
+
+  ~EventTimer() {
+    hipError_t error;  // to avoid compiler warnings
+
+    error = hipEventRecord(stop_, GetStream());
+    error = hipEventSynchronize(stop_);
+
+    float ms;
+    error = hipEventElapsedTime(&ms, start_, stop_);
+    Record(ms);
+
+    error = hipEventDestroy(start_);
+    error = hipEventDestroy(stop_);
+  }
+
+ private:
+  hipEvent_t start_;
+  hipEvent_t stop_;
+};
+
+class CpuTimer : public Timer {
+ public:
+  CpuTimer(float& time, hipStream_t stream = nullptr) : Timer(time, stream) {
+    start_ = std::chrono::steady_clock::now();
+  }
+
+  ~CpuTimer() {
+    hipError_t error;  // to avoid compiler warnings
+    error = hipStreamSynchronize(GetStream());
+
+    stop_ = std::chrono::steady_clock::now();
+
+    std::chrono::duration<float, std::milli> ms = stop_ - start_;
+    Record(ms.count());
+  }
+
+ private:
+  std::chrono::time_point<std::chrono::steady_clock> start_;
+  std::chrono::time_point<std::chrono::steady_clock> stop_;
+};
+
+template <typename Derived> class Benchmark {
+ public:
+  Benchmark()
+      : iterations_(cmd_options.iterations),
+        warmups_(cmd_options.warmups),
+        display_output_(!cmd_options.no_display),
+        progress_bar_(cmd_options.progress) {
+    benchmark_name_ = Catch::getResultCapture().getCurrentTestName();
+  }
+
+  Benchmark(const Benchmark&) = delete;
+  Benchmark& operator=(const Benchmark&) = delete;
+
+  static constexpr ssize_t kWarmup = -1;
+
+  void Configure(size_t iterations, size_t warmups) {
+    iterations_ = iterations;
+    warmups_ = warmups;
+  }
+
+  void AddSectionName(const std::string& section_name) { benchmark_name_ += "/" + section_name; }
+
+  using ModifierSignature = std::function<float(float)>;
+  void RegisterModifier(const ModifierSignature& modifier) { modifier_ = modifier; }
+
+  template <typename... Args> std::tuple<float, float, float, float> Run(Args&&... args) {
+    AddSectionName(std::to_string(iterations_));
+    AddSectionName(std::to_string(warmups_));
+
+    auto& derived = static_cast<Derived&>(*this);
+
+    current_ = kWarmup;
+    for (size_t i = 0u; i < warmups_; ++i) {
+      PrintProgress("warmup", static_cast<int>(100.f * (i + 1) / warmups_));
+      derived(args...);
+    }
+    time_ = .0;
+
+    std::vector<float> samples;
+    samples.reserve(iterations_);
+
+    for (current_ = 0; current_ < iterations_; ++current_) {
+      PrintProgress("measurement", static_cast<int>(100.f * (current_ + 1) / iterations_));
+      derived(args...);
+      if (modifier_) time_ = modifier_(time_);
+      samples.push_back(time_);
+      time_ = .0;
+    }
+
+    float sum = std::reduce(cbegin(samples), cend(samples));
```

Comment:
error: no member named 'reduce' in namespace 'std'

Reply:
@mangupta: Can you please check if uploaded patch resolves build error? Previously mentioned patch is available within commit https://github.com/ROCm-Developer-Tools/hip-tests/pull/117/commits/dd4e5e9f5bfb1e2b89da60263f0b3b1c93a0886f.